### PR TITLE
Mutual friendships

### PR DESCRIPTION
**Pull request by tzvaita**
The mutual friendships feature was made complete as follows:
- If a user creates a friendship it will now show on the requested user account
- If the request is accepted another friendship is created in the friendship table with a friendship status of accepted/true.

### CODE REVIEW

**Comment by Anna-Myzukina** [comment]
# Changes required  ♻️ 

### Hello @tzvaita   🌟

**Very nice work on this Milestone 6: Friendships v2, however, the following issues need to be fixed before moving on ⬇️** 

- I get double flash messages on each page.









### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
   <% if friends_already(@user) %>
-    <%= button_to 'Remove',friendship_path(finding_friendship_id(@user)), method: :delete %>
+    <%= button_to 'Remove',friendship_path(find_friendship_id(@user)), method: :delete %>
   <% else %>
   <span class="profile-link">
     <%= button_to  'Invite to friendship', friendships_path({:friend_id => @user.id}), method: :post %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb:15]: That section part of `if..else` statement implies that a connected user will be able to **Invite** himself to friendship

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required  :red_circle: 

Hello,

Thanks for the improvements! 
However, there are some bugs to fix.

1. It was asked [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png), to include posts written by logged-in user and their friends not all the posts

2. You can't invite to **Friendship** the connected user himself.

3. The whole friendship process is working. However, **Remove** the friendship deletes only online line.
**BEFORE**
```ruby
2.7.0 :001 > Friendship.all
  Friendship Load (0.4ms)  SELECT  "friendships".* FROM "friendships" LIMIT $1  [["LIMIT", 11]]
 => #<ActiveRecord::Relation [#<Friendship id: 2, user_id: 2, friend_id: 1, confirmed: true, created_at: "2020-05-11 05:21:35", updated_at: "2020-05-11 05:22:21">, #<Friendship id: 3, user_id: 1, friend_id: 2, confirmed: true, created_at: "2020-05-11 05:22:21", updated_at: "2020-05-11 05:22:21">]> 
2.7.0 :002 >
```

**AFTER**
```ruby
2.7.0 :001 > Friendship.all
  Friendship Load (0.3ms)  SELECT  "friendships".* FROM "friendships" LIMIT $1  [["LIMIT", 11]]
 => #<ActiveRecord::Relation [#<Friendship id: 2, user_id: 2, friend_id: 1, confirmed: true, created_at: "2020-05-11 05:21:35", updated_at: "2020-05-11 05:22:21">]> 
2.7.0 :002 > 
```

Kindly make the changes and submit a new code review request.
Thanks.

Mael FOSSO

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes Required :muscle: 
Hi there team @tzvaita :hand: 

Great job implementing the requested changes :clap: :clap: :clap: 
But before moving on we need to fix a bug on your app.

- [ ] Right now, a friendship invitation can't be deleted. Neither by the user who sent it or the user that received it. The application throws an error when I tried to do that.
Screenshot from 2020-05-11 20-39-34

Please kindly fix that and submit for code review again.
Feel free to message me on slack if you have any questions

Happy coding :computer: 
**Sergio Torres**

**Review by jofer86 — CHANGES_REQUESTED** [review]
## :wave: :wave: Hello!, I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- It seems that no more changes were done after @Torres-ssf review, please do tend to the required changes and ask for a new code review.
--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @tzvaita :handshake: 

### Thank you for doing the changes requested by the previous reviewer :medal_sports: your application doesn't show the deleting request error but is still having faulty behavior that needs to be fixed :construction: :construction_worker: 

#### Faulty behaviour

- [ ] When a user(sender) sends a friend request to another user(receiver) and then cancels the friend request this friend request is not being deleted in the database so when the user(sender) sends another friend request to the same user(receiver) and the user(receiver) accepts the request, your application still shows that a friend request is pending let's fix this:

Example:

Maru sends a friend request to Audrey and then canceled it, and then sent another friend request to Audrey again.

Audrey accepted the friend request and now Maru and Audrey are friends Audrey can see Maru's posts in her TIMELINE tab
image

But when Audrey goes to the ALL USERS tab there is still a pending friend request from Maru
image

#### Unit Integrations test

There should be a test case that validates that after a Friendship is created is not possible to create another one from the same user_id and friend_id pair, but your application is letting it:

image

* [ ] I think your application doesn't have enough Unit test cases for friendships ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) and testing can help you solve  a lot of faulty behavior so let's add the following test cases:

- Unit Test cases:
  - validates against duplicate friendship
  - validates against duplicate inverse-friendship
  - validates against self-friendship

### That's it! :raised_hands: 

### Please :pray: do these changes :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by nikasulo — APPROVED** [review]
# Great work @tzvaita 
Please merge and keep moving forward! I wish you the very best and see you at the top!

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Pull request merged**